source/slang/slang-check-decl.cpp

```diff
@@ -3248,9 +3253,10 @@ bool isProperConstraineeType(Type* type)
     return true;
 }
 
-void SemanticsDeclHeaderVisitor::validateGenericConstraintSubType(
+bool SemanticsDeclHeaderVisitor::validateGenericConstraintSubType(
     GenericTypeConstraintDecl* decl,
-    TypeExp type)
+    TypeExp type,
+    bool diagnose)
```

Comment:
Pass in DiagnosticSink* instead of a bool, and diagnose using the passed in sink if it is not null.